src/pdc-api.ts

```diff
@@ -85,12 +85,13 @@ interface Proposals {
   total: number;
 }
 
-const useProposals = (page: string, count: string) => (
+const useProposals = (page: string, count: string, query: string) => (
```

Comment:
Could we default to `query = ''` so this can be an optional parameter?

In fact, I think it makes sense for `page` to be optional (defaulting to `1`), too. For practical purposes (i.e. discouraging countless queries) it makes sense to have a default `count`.

What do you think?

Reply:
One or two optional parameters can be helpful, but making all three optional starts to be confusing!

Right now, there are only two call sites: `<ProposalListLoader>` and `<ProposalListGridLoader>`. The latter currently has placeholder values, but once we figure out how we want to do sidebar search & pagination, that'll change.

In `<ProposalListLoader>`, and eventually in `<ProposalListGridLoader>`, we'll always be pulling out the search params from the data-viewer URL and passing them along to `useProposals`, so I don't think we'd ever take advantage of the optional parameter. Passing the possibly-`null` values returned by `searchParams.get('key')` into this hook complicates the hook somewhat.

In general, I think I'd prefer to initialize default values as close to when we name their variables as possible. Passing around "missing" values is harder (for me) to think about, and makes working with such values more difficult.

That said, something you might have been hinting at is that we should have a single constant for each default value, rather than repeating magic numbers (which are strings, in this case). Again, the call in `<ProposalListGridLoader>` is some intentional tech debt, but that's no reason not to start naming and using constants now! I will fix.

@reefdog is that at all convincing?

Reply:
I did _also_ think about specifying the defaults as constants, so I'm glad you did that!

I was more thinking the ergonomics of calling `useProposals()` and how, even if we will always be invoking it with all three params, there's nothing inherently required about any of them (that I'm aware of). Still, I'm happy to let this one go and call it resolved! I do understand your logic and am not overly concerned.